jenkins/Jenkinsfile-blossom.premerge

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ * Jenkinsfile for building rapids-plugin on blossom
+ *
+ */
+
+@Library('shared-libs') _
+@Library('blossom-github-lib@master')
+import ipp.blossom.*
+
+
+def pluginDockerBuild = """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: plugin-docker
+    image: docker:19.03.12
+    command:
+    - sleep
+    args:
+    - 99d
+    resources:
+      requests:
+        cpu: 4
+        memory: 8Gi
+    env:
+      - name: DOCKER_HOST
+        value: tcp://localhost:2375
+  - name: plugin-docker-daemon
+    image: docker:19.03.12-dind
+    securityContext:
+      privileged: true
+    env:
+      - name: DOCKER_TLS_CERTDIR
+        value: ""
+  nodeSelector:
+    kubernetes.io/os: linux
+"""
+
+def pluginPremerge = """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: plugin-premerge
+    image: IMAGE_PREMERGE
+    imagePullPolicy: Always
+    resources:
+      requests:
+        cpu: 8
+        memory: 16Gi
+      limits:
+        nvidia.com/gpu: 1
+    restartPolicy: Never
+    backoffLimit: 2
+    tty: true
+  nodeSelector:
+    kubernetes.io/os: linux
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key: nvidia.com/gpu_type
+            operator: In
+            values:
+            - Tesla_V100S_PCIE_32GB
+            - Tesla_V100_PCIE_32GB
+            - TITAN_RTX
+            - Tesla_T4
+"""
+
+def githubHelper // blossom github helper
+def IMAGE_PREMERGE // temp image for premerge test
+def PREMERGE_TAG
+def skipped = false
+
+pipeline {
+    agent {
+        kubernetes {
+            label "premerge-init-${BUILD_TAG}"
+            cloud 'sc-ipp-blossom-prod'
+        }
+    }
+
+    options {
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '50'))
+        skipDefaultCheckout true
+        timeout(time: 180, unit: 'MINUTES')
+    }
+
+    environment {
+        JENKINS_ROOT = 'jenkins'
+        PREMERGE_SCRIPT = '$JENKINS_ROOT/spark-premerge-build.sh'
+        MVN_URM_MIRROR = '-s jenkins/settings.xml -P mirror-apache-to-urm'
+        LIBCUDF_KERNEL_CACHE_PATH = '/tmp/.cudf'
+        ARTIFACTORY_NAME = "${ArtifactoryConstants.ARTIFACTORY_NAME}"
+        GITHUB_TOKEN = credentials("github-token")
+        URM_CREDS = credentials("svcngcc_artifactory")
```

Comment:
this perhaps goes to my previous question are we using svcngcc user?

Reply:
no, we are not using any svcngcc user anymore on blossom. We keep the same credential name to try make changes at small as possible during the premerge migration period, and we should change this when we shutdown old premerge pipeline on ngcc.

Reply:
ok